geps/gep-1619.md

```diff
@@ -62,14 +62,110 @@ Header-based stateful sessions are achieved by a backend or gateway providing an
 
 Session information can be also encoded into the request URL to establish a persistent session. The server rewrites the client's URL to encode the new session information automatically. The server then decodes the session information from the URL to identify the session.
 
-**Session Initiation**
+### Session Persistence Initiation
 
-For both header-based and cookie-based sessions, either the gateway or the backend server can initiate establishing the session via the appropriate header or set-cookie attributes. The following rules apply based on who initiates the session:
+For both header-based and cookie-based sessions, the initiation of session persistence is possible from various sources,
+including the gateway, backend, a sidecar in a backend, or any other infrastructure component. The client application
+also plays in a role in session initiation. If a client doesn't want a request to belong an existing session, it can
+remove the session cookie or identifier to signal the gateway or backend to recreate the session.
 
+Let's focus on the two primary sources: gateways and the backend. The following rules apply based on who initiates the session:
+- If the gateway initiates the session, the backend will be presented with session attributes regardless if it enabled them.
 - If the backend initiates the session, the gateway should allow this and not force persistence connections, unless specifically configured to. The gateway may decode and alter the cookie established by the backend to achieve session persistence.
-- If the gateway initiates the session, the backend will be presented with session attributes regardless if it enabled them. 
 
-The client application also plays in a role in session initiation. If a client doesn't want a request to belong an existing session, it can remove the session cookie or identifier to signal the gateway or backend to recreate the session.
+In the next sections, we will take a closer look at the initiation of sessions in both the gateway and the backend.
+Please note that in the next sections, we are examining the various scenarios in which a session can be initiated. We
+are not prescribing specific implementations for session persistence. The intention is to understand the possibilities
+and behaviors related to session initiation while the [API](#api) section will provide more details on specific
+implementation details.
+
+#### Gateway Initiated Sessions
+
+This represents the most straightforward scenario for utilizing cookies. When a request is made, the gateway includes
+the `set-cookie` header in the final response, prompting the client to generate a cookie. This cookie is subsequently
```

Comment:
'prompting the client to store the cookie'. Cookie is generated by gateway. 

Q: does the backend knows anything about the gateway-generated cookie ( is it sent to backend too in the first request) ? And I assume backends using affinity may set their own cookies too, java for sure would.